### DISCUSSION TITLE
DataGrid completely invisible

### DISCUSSION THREAD

**Discussion by PerfectlyFineCode** [AvaloniaUI/Avalonia#8424] [Q&A] [answered]
**Describe the bug**
DataGrid does not render at all

**To Reproduce**
1.) Create an empty Window (AXAML)
2.) Add Grid component
3.) Inside of Grid Component, add DataGrid.
4.) It should not render at all.

**Expected behavior**
It should be visible and is rendering.

**Screenshots**
![image](https://user-images.githubusercontent.com/70201838/176760306-5504353f-30ed-4c6e-9a2b-cd0bb48c0c04.png)

**Desktop (please complete the following information):**

- OS: Windows 11
- OS Build: 22000.739

**Avalonia Version**
Tested on all three versions, same thing on all of them.
0.10.15, 0.10.14, 0.10.12

**Comment by wieslawsoltes** [comment, marked as answer]
You need to add styles for DataGrid as it's separate control not included in themes 

https://docs.avaloniaui.net/docs/controls/datagrid#add-required-styles-to-app.axaml

```xaml
<Application.Styles>
    <FluentTheme Mode="Light" />
    <StyleInclude Source="avares://Avalonia.Controls.DataGrid/Themes/Fluent.xaml"/>
</Application.Styles>
```

**Reply by timunie** [reply]
And you need to add a nuget package for it separately for release build. Debug works without referencing it which is somehow confusing, but there is no way to change it.

**Reply by FoggyFinder** [reply]
> And you need to add a nuget package for it separately for release build. Debug works without referencing it which is somehow confusing, but there is no way to change it.

In short, it happens because of `Avalonia.Diagnostics` package

```xml
<PackageReference Condition="'$(Configuration)' == 'Debug'" Include="Avalonia.Diagnostics" Version="0.10.14" />
 ```

which depends on `Avalonia.Controls.DataGrid` one.

**Reply by PerfectlyFineCode** [reply]
If so, that's very interesting.

I'm so used to WPF, where all available controls are available by default and already includes a standard style.
In my opinion it definitely shouldn't be like this, that you have to include a style manually to be able to see the control in the first place.

**Reply by timunie** [reply]
The thing is that DataGrid is a heavy control that will blow up your app size so the decision was to not include it by default and have the user to opt in if needed as not everyone needs it. 

In the end it may get philosophical now. We have to deal with it as it I guess.
